### DISCUSSION TITLE
How can I get the current timestamp minus 30 minutes in nushell?

### DISCUSSION THREAD

**Discussion by Tom-Boscher** [nushell/nushell#10469] [Q&A] [answered]
I searched for a replacement of `date -d` in Nushell but I didn't find anything. How can I get the current datetime minus N minutes ? Like this `$((date now) - 30 minutes)`

**Comment by hustcer** [comment, marked as answer]
```Nushell
(date now) - 30min
```

**Reply by Tom-Boscher** [reply]
Thank you very much, I didn't expect it was so simple!

**Reply by Tom-Boscher** [reply]
Is it possible to convert it into UNIX epoch time?

**Reply by panicbit** [reply]
Yes, with `format date`:
```nu
(date now) - 30min | format date %s
```

**Reply by Tom-Boscher** [reply]
Thanks! Last question, how can I use user input in (date now) - 30min?

I try this but it doesn't work. Sorry if this seems trivial, I just began to learn Nushell last days.
```
let selected_duration = input ("Log duration:") | into int
let duration_logs = (date now) - ($selected_duration)min
```

**Reply by amtoine** [reply]
@Tom-Boscher 
what about the following? :yum: 
```nushell
let selected_duration = input ("Log duration:") | into int | into duration
let duration_logs = (date now) - $selected_duration
```

> **Note**
> the user input would have to be in nanoseconds

you can also do
```nushell
let selected_duration = input ("Log duration:") | into duration
```
but the user will have to type a duration unit, e.g. `123sec`, without adding any extra logic to add a default unit :wink:

**Reply by Tom-Boscher** [reply]
Thank you! I did `let selected_duration = (input ("Log duration (in minutes):") | into int | into duration) * 60000000000` and it works like a charm 😊

**Reply by amtoine** [reply]
noishe :yum: 

i would say the following might be a bit easier than the super huge number :wink: 
```nushell
input ("Log duration (in minutes):") | into int | $in * 1min
```
